chapter.md

```diff
@@ -0,0 +1,735 @@
+  name: undefined,
+  bioSummary: {
+    age: undefined,
+    size: undefined,
+    alignment: undefined,
+    race: undefined,
+  },
+  skills: []
+};
+
+const initialState: IForm = {
+  character: characterInitialState
+};
+```
+
+We could fold the definition of `characterInitialState` into `initialState`
+to create one large literal object,
+but again,
+since we're likely to add more state information
+(like the equipment the character is carrying),
+we have decided that `initialState` will only ever be
+a list of top-level objects conforming to separately-defined interfaces.
+
+> We have defined our initial state in a file of its own
+> to make it easier to find and update.
+> In a real application,
+> it would probably be created by some sort of factory
+> so that we could swap in something else for testing.
+> Note that in real life, it would *not* come from a database:
+> instead, it's the blank slate that would be updated from a database
+> as the application is being bootstrapped.
+> And yes,
+> that bootstrap process would be implemented as
+> a series of update actions.
+
+Another early decision is that we will define functions to create actions,
+and that each action will have two elements:
+a string called `type` (which Redux requires),
+and a sub-object called `payload` that will contain
+the path to the part of the store it applies to
+and whatever extra values are needed to update the state.
+Redux doesn't mandate this,
+but again,
+experience teaches us that creating literal objects
+in many different places throughout our program
+is just as risky as using magic numbers
+rather than defining a constant and referring to it.
+
+As an example of an action creation function,
+here's one that saves a form's value:
+
+```ts
+const saveForm = (path, value) => ({
+  type: 'SAVE_FORM',
+  payload: {
+    path: path,
+    value: value
+  }
+});
+```
+
+Now that we know the shapes of our store and our action,
+the first version of the reducer is easy to write.
+To keep it short,
+we rely on three helper functions from [Ramda][ramda-site],
+a library of functional utilities for JavaScript:
+
+*   `assocPath`: makes a shallow clone of an object,
+    replacing the specified property with the specified value as it does so.
+    (Think of this as "make me a copy of X, but with Y set to Z".)
+*   `lensPath`: returns a *lens* that points to a particular location in a nested data structure.
+    (This is like a path in a filesystem, and is used for similar purposes.)
+*   `merge`: create a copy of one object with properties merged in from a second object.
+    (This is like `assocPath`, but using a second object to get multiple changes at once.)
+
+With these helpers in hand,
+our `formReducer` looks like this:
+
+```ts
+import { lensPath, assocPath, merge } from 'ramda';
+
+export function formReducer(state = initialState: IForm, action) {
+  switch (action.type) {
+
+  case 'SAVE_FORM':
+    const lensForProp = lensPath(action.payload.path);
```

Comment:
I will update our repo and the code to use path instead, so this switch case can be easily rewritten using `path` from ramda, which is like a `lensPath` + `view`, something liek:

```ts
 case 'SAVE_FORM':
     return assocPath(
       action.payload.path,
       merge(path(action.payload.path, state), action.payload.value),
       state
     );
```
might be easier for the reader to grasp